### PR TITLE
KAZOO-4950: process kazoo modules for raw jobj

### PR DESCRIPTION
**Pull request by jamesaimonetti**

### CODE REVIEW

File: applications/call_inspector/src/parsers/ci_parser_kamailio.erl [hunk 1]
```diff
@@ -282,7 +282,7 @@ cleanse_data_and_get_timestamp(Data0) ->
 -spec cleanse_data_fold({ne_binary() | kz_now()} | ne_binary()
                        ,cleanse_acc()
                        ) -> cleanse_acc().
-cleanse_data_fold({RawTimestamp}, {Acc, TS}) ->
+cleanse_data_fold({'timestamp', RawTimestamp}, {Acc, TS}) ->
```
Comment (fenollp) [inline applications/call_inspector/src/parsers/ci_parser_kamailio.erl]: Your check is too naive: this is not a JObj. BTW spec needs to be updated with your changes.

File: applications/ecallmgr/src/ecallmgr_fs_node_sup.erl [hunk 2]
```diff
@@ -31,11 +31,16 @@
...
+-define(CHILDREN, [<<"authn">>
+                  ,<<"channel">>
                   ,<<"conference">>
+                  ,<<"config">>
                   ,<<"event_stream_sup">>
                   ,<<"msg">>
+                  ,<<"node">>
+                  ,<<"notify">>
+                  ,<<"resource">>
+                  ,<<"route">>
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_node_sup.erl]: Changing the order is OK here?

File: core/kazoo_number_manager/src/knm_phone_number.erl [hunk 3]
```diff
@@ -520,7 +520,8 @@ features_list(N) ->
     sets:to_list(sets:from_list(kz_json:get_keys(features(N)))).
 
 -spec set_features(knm_phone_number(), kz_json:object()) -> knm_phone_number().
-set_features(N, Features=?JSON_WRAPPER(_)) ->
+set_features(N, Features) ->
+    'true' = kz_json:is_json_object(Features),
```
Comment (fenollp) [inline core/kazoo_number_manager/src/knm_phone_number.erl]: Isn't pattern checking a macro value way less expensive? What is wrong with the macro?
Also if you remove the macro please remove the now useless include_lib(kz_json.hrl).

File: core/kazoo_config/src/kapps_config.erl [hunk 4]
```diff
@@ -102,17 +102,24 @@ get_binary(Category, Key, Default, Node) ->
...
-        'undefined' -> 'undefined';
-        ?JSON_WRAPPER(_)=V -> V;
-        _Else -> 'undefined'
+    V = get(Category, Key),
+    as_json_value(V).
+
+-spec as_json_value(any()) -> api_object().
+as_json_value('undefined') -> 'undefined';
+as_json_value(V) ->
+    case kz_json:is_json_object(V) of
```
Comment (fenollp) [inline core/kazoo_config/src/kapps_config.erl]: Same here wrt the macro. No without: code is now more bloated

File: core/kazoo_ast/src/cb_api_endpoints.erl [hunk 5]
```diff
@@ -42,7 +42,7 @@ api_to_ref_doc(Module, Paths, ?CURRENT_VERSION) ->
     DocPath = ?REF_PATH(BaseName),
     'ok' = file:write_file(DocPath, Doc);
 api_to_ref_doc(_Module, _Paths, _Version) ->
-    io:format("skipping ~p version ~p~n", [_Module, _Version]).
+    'ok'.
```
Comment (fenollp) [inline core/kazoo_ast/src/cb_api_endpoints.erl]: Why remove such information? On the contrary I believe this is worth mentioning to the user: "no v1 endpoints end up in swagger.json".

File: Makefile [hunk 6]
```diff
@@ -172,6 +172,7 @@ $(FMT):
 
 fmt: TO_FMT ?= $(shell find applications core -iname '*.erl' -or -iname '*.hrl' -or -iname '*.app.src')
 fmt: $(FMT)
+	@ERL_LIBS=deps/:core/:applications/ $(ROOT)/scripts/no_raw_json.escript
```
Comment (fenollp) [inline Makefile]: I'd say this should go into a `code_checks` target which you should create.
`no_raw_json` doesn't write to files, does it?

**Comment by fenollp** [comment]
BTW your build fails in part due to trying to run your escript before Kazoo is even compiled

File: applications/call_inspector/src/parsers/ci_parser_kamailio.erl (hunk 1 above)
Comment (jamesaimonetti) [inline applications/call_inspector/src/parsers/ci_parser_kamailio.erl]: Well, yes and no. It would have required a lot more effort to figure out if the variable inside the `{...}` was a list or not. This was an easier fix, and tagged tuples are more idiomatic, I believe. Why wrap a timestamp in a 1-tuple to being with?

File: applications/ecallmgr/src/ecallmgr_fs_node_sup.erl (hunk 2 above)
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_node_sup.erl]: AFAIK none of the supervisors rely on the others being up yet. It is a one-for-one supervisor so they should operate independently anyway.

File: core/kazoo_number_manager/src/knm_phone_number.erl (hunk 3 above)
Comment (jamesaimonetti) [inline core/kazoo_number_manager/src/knm_phone_number.erl]: I believe the macro usage was added to pacify Dialyzer but I could be mis-remembering.

File: core/kazoo_ast/src/cb_api_endpoints.erl (hunk 5 above)
Comment (jamesaimonetti) [inline core/kazoo_ast/src/cb_api_endpoints.erl]: If you deduplicated all the logging, perhaps. As it was, it was quite chatty and offered little value imo.

Comment (fenollp) [inline core/kazoo_ast/src/cb_api_endpoints.erl]: I agree it is chatty but not that much. Anyway these are unrelated changes, could you move them to a different PR?
